Source/Renderer/ComputeEngine.js

```diff
@@ -0,0 +1,210 @@
+        if (!defined(vertexArray)) {
+            var geometry = new Geometry({
+                attributes : {
+                    position : new GeometryAttribute({
+                        componentDatatype : ComponentDatatype.FLOAT,
+                        componentsPerAttribute : 2,
+                        values : [
+                            -1.0, -1.0,
+                            1.0, -1.0,
+                            1.0,  1.0,
+                            -1.0,  1.0
+                        ]
+                    }),
+
+                    textureCoordinates : new GeometryAttribute({
+                        componentDatatype : ComponentDatatype.FLOAT,
+                        componentsPerAttribute : 2,
+                        values : [
+                            0.0, 0.0,
+                            1.0, 0.0,
+                            1.0, 1.0,
+                            0.0, 1.0
+                        ]
+                    })
+                },
+                // Workaround Internet Explorer 11.0.8 lack of TRIANGLE_FAN
+                indices : new Uint16Array([0, 1, 2, 0, 2, 3]),
+                primitiveType : PrimitiveType.TRIANGLES
+            });
+
+            vertexArray = VertexArray.fromGeometry({
+                context : computeEngine._context,
+                geometry : geometry,
+                attributeLocations : viewportQuadAttributeLocations,
+                bufferUsage : BufferUsage.STATIC_DRAW,
+                interleave : true
+            });
+
+            computeEngine._viewportQuadVertexArray = vertexArray;
+        }
+
+        return vertexArray;
+    }
+
+    function createFramebuffer(context, outputTexture) {
+        var fbo = new Framebuffer({
+            context : context,
+            colorTextures : [outputTexture],
+            destroyAttachments : false
+        });
+        return fbo;
+    }
+
+    function createViewportQuadShader(context, fragmentShaderSource) {
+        return ShaderProgram.fromCache({
+            context : context,
+            vertexShaderSource : ViewportQuadVS,
+            fragmentShaderSource : fragmentShaderSource,
+            attributeLocations : viewportQuadAttributeLocations
+        });
+    }
+
+    function createRenderState(width, height) {
+        if ((!defined(renderStateScratch)) ||
+            (renderStateScratch.viewport.width !== width) ||
+            (renderStateScratch.viewport.height !== height)) {
+
+            renderStateScratch = RenderState.fromCache({
+                viewport : new BoundingRectangle(0, 0, width, height)
+            });
+        }
+        return renderStateScratch;
+    }
+
+    ComputeEngine.prototype.execute = function(computeCommand) {
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(computeCommand)) {
+            throw new DeveloperError('computeCommand is required.');
+        }
+        //>>includeEnd('debug');
+
+        if (defined(computeCommand.preExecute)) {
```

Comment:
This is before the potential `DeveloperError`s below because it might modify the command, right?  Let's add a comment since so in the future we don't think "why not put all all DeveloperErrors at the top together?"